packages/shared/friends/sagas.ts

```diff
@@ -26,158 +24,166 @@ import {
-const MESSAGE_LIFESPAN_MILLIS = 1000
 
-const SEND_STATUS_INTERVAL_MILLIS = 5000
-type PresenceMemoization = { realm: SocialRealm | undefined; position: UserPosition | undefined }
-const presenceMap: Record<string, PresenceMemoization | undefined> = {}
+const INITIAL_CHAT_SIZE = 50
+const MESSAGE_LIFESPAN_MILLIS = 1_000
+const SEND_STATUS_INTERVAL_MILLIS = 60_000
+const MIN_TIME_BETWEEN_FRIENDS_INITIALIZATION_RETRIES_MILLIS = 1000
+const MAX_TIME_BETWEEN_FRIENDS_INITIALIZATION_RETRIES_MILLIS = 256000
 
 export function* friendsSaga() {
-  if (WORLD_EXPLORER) {
-    // We don't want to initialize the friends & chat feature if we are on preview or builder mode
-    yield takeEvery(USER_AUTHENTIFIED, initializeFriendsSaga)
-
-    yield takeEvery(UPDATE_FRIENDSHIP, trackEvents)
-    yield takeEvery(UPDATE_FRIENDSHIP, handleUpdateFriendship)
-    yield takeEvery(SEND_PRIVATE_MESSAGE, handleSendPrivateMessage)
-  }
+  // We don't want to initialize the friends & chat feature if we are on preview or builder mode
+  yield fork(initializeFriendsSaga)
+  yield fork(initializeStatusUpdateInterval)
+  yield fork(initializeReceivedMessagesCleanUp)
+
+  yield takeEvery(SET_MATRIX_CLIENT, configureMatrixClient)
+  yield takeEvery(UPDATE_FRIENDSHIP, trackEvents)
+  yield takeEvery(UPDATE_FRIENDSHIP, handleUpdateFriendship)
+  yield takeEvery(SEND_PRIVATE_MESSAGE, handleSendPrivateMessage)
 }
 
-function* initializeFriendsSaga(action: UserAuthentified) {
-  if (!action.payload.isGuest) {
-    yield call(waitForRealmInitialized)
-
-    try {
-      const synapseUrl: string = yield select(getSynapseUrl)
-      yield call(initializePrivateMessaging, synapseUrl, action.payload.identity)
-    } catch (e) {
-      logger.error(`error initializing private messaging`, e)
+function* initializeFriendsSaga() {
+  let secondsToRetry = MIN_TIME_BETWEEN_FRIENDS_INITIALIZATION_RETRIES_MILLIS
 
-      yield call(waitForRendererInstance)
+  while (true) {
+    yield race({
+      auth: take(USER_AUTHENTIFIED),
+      delay: delay(secondsToRetry)
+    })
 
-      getUnityInstance().ConfigureHUDElement(HUDElementID.FRIENDS, { active: false, visible: false })
-      getUnityInstance().ShowNotification({
-        type: NotificationType.GENERIC,
-        message: 'There was an error initializing friends and private messages',
-        buttonMessage: 'OK',
-        timer: 7
-      })
-      trackEvent('error', {
-        context: 'kernel#saga',
-        message: 'There was an error initializing friends and private messages',
-        stack: ''
-      })
+    yield call(waitForRealmInitialized)
+    yield call(waitForRendererInstance)
+
+    const currentIdentity: ExplorerIdentity | undefined = yield select(getCurrentIdentity)
+    const client: SocialAPI | null = yield select(getSocialClient)
+    const isLoggedIn: boolean = (currentIdentity && client && (yield apply(client, client.isLoggedIn, []))) || false
+
+    const shouldRetry = !isLoggedIn
+
+    if (shouldRetry) {
```

Comment:
Should we have a limited number of retries instead of doing it indefinitely?

Reply:
The idea was to have it indefinitely recovery. but the client has a problem that prevents it from happening if some events are pending to be sent to matrix. I'd recommend to stop using the dcl-social-api and using matrix-js-sdk from kernel directly instead